pkg/api/v1alpha1/notifications_target.go

```diff
@@ -0,0 +1,462 @@
+	if err != nil {
+		r.Logger.Error("error fetching notification targets", zap.Error(err))
+		sendError(c, http.StatusBadRequest, "error listing notification targets: "+err.Error())
+
+		return
+	}
+
+	c.JSON(http.StatusOK, notificationTargets)
+}
+
+// createNotificationTarget creates a notification target in DB
+func (r *Router) createNotificationTarget(c *gin.Context) {
+	req := &NotificationTargetReq{}
+	if err := c.BindJSON(req); err != nil {
+		sendError(c, http.StatusBadRequest, "unable to bind request: "+err.Error())
+		return
+	}
+
+	if req.Name == "" {
+		sendError(c, http.StatusBadRequest, "notification target name is required")
+		return
+	}
+
+	if req.Description == "" {
+		sendError(c, http.StatusBadRequest, "notification target description is required")
+		return
+	}
+
+	notificationTarget := &models.NotificationTarget{
+		Name:           req.Name,
+		Description:    req.Description,
+		DefaultEnabled: req.DefaultEnabled,
+	}
+
+	notificationTarget.Slug = slug.Make(notificationTarget.Name)
+
+	tx, err := r.DB.BeginTx(c.Request.Context(), nil)
+	if err != nil {
+		sendError(c, http.StatusBadRequest, "error starting notification target create transaction: "+err.Error())
+		return
+	}
+
+	if err := notificationTarget.Insert(c.Request.Context(), tx, boil.Infer()); err != nil {
+		msg := fmt.Sprintf("error creating notification target: %s", err.Error())
+
+		if err := tx.Rollback(); err != nil {
+			msg += fmt.Sprintf("error rolling back transaction: %s", err.Error())
+		}
+
+		sendError(c, http.StatusBadRequest, msg)
+
+		return
+	}
+
+	event, err := dbtools.AuditNotificationTargetCreated(
+		c.Request.Context(),
+		tx,
+		getCtxAuditID(c),
+		getCtxUser(c),
+		notificationTarget,
+	)
+	if err != nil {
+		msg := fmt.Sprintf("error creating notification target (audit): %s", err.Error())
+
+		if err := tx.Rollback(); err != nil {
+			msg += fmt.Sprintf("error rolling back transaction: %s", err.Error())
+		}
+
+		sendError(c, http.StatusBadRequest, msg)
+
+		return
+	}
+
+	if err := updateContextWithAuditEventData(c, event); err != nil {
+		msg := fmt.Sprintf("error creating notification target: %s", err.Error())
+
+		if err := tx.Rollback(); err != nil {
+			msg += fmt.Sprintf("error rolling back transaction: %s", err.Error())
+		}
+
+		sendError(c, http.StatusBadRequest, msg)
+
+		return
+	}
+
+	if err := tx.Commit(); err != nil {
+		msg := fmt.Sprintf("error committing notification target create: %s", err.Error())
+
+		if err := tx.Rollback(); err != nil {
+			msg += fmt.Sprintf("error rolling back transaction: %s", err.Error())
+		}
+
+		sendError(c, http.StatusBadRequest, msg)
+
+		return
+	}
+
+	// CRDB cannot refresh a materialized view inside explicit transactions
+	// https://www.cockroachlabs.com/docs/stable/views#known-limitations
+	if err := dbtools.RefreshNotificationDefaults(c.Request.Context(), r.DB); err != nil {
+		sendError(c, http.StatusInternalServerError, err.Error())
+		return
+	}
```

Comment:
are you calling this because the view wont get updated at all after the transaction is committed? my read of the link is just that you can't do this *inside* a transaction.